claat/render/html.go

```diff
@@ -224,18 +224,23 @@ func (hw *htmlWriter) button(n *types.ButtonNode) {
 }
 
 func (hw *htmlWriter) code(n *types.CodeNode) {
-	hw.writeString("<pre>")
-	if !n.Term {
-		hw.writeString("<code")
-		if n.Lang != "" {
-			hw.writeFmt(" language=%q class=%q", n.Lang, n.Lang)
-		}
-		hw.writeBytes(greaterThan)
+	hw.writeString(`<pre class="code-block">`)
+
+	hw.writeString(`<div class="code-menu">`)
+	hw.writeString(`<input type="button" onclick="wrap(this);" value="&#8677;">`)
+	hw.writeString(`<input type="button" onclick="copy(this);" value="&#10697;">`)
+	hw.writeString(`</div>`)
```

Comment:
We prefer to do the heavy lifting in custom elements rather than in inline JavaScript. Inline event handlers are also disallowed by Chrome's Content Security Policy[1]. For the 4 lines above, could we instead just write a `<google-codelab-code-menu>` custom element to the DOM, then create a new custom element with the code that you have inlined in template.html below?

[1] https://developer.chrome.com/extensions/contentSecurityPolicy#JSExecution

Reply:
Thanks @shawnbuso. Good suggestion.  Let me refactor my code...